UnitTests/CommonTestUtils/SubmoduleTestHelpers.cs

```diff
@@ -28,14 +31,14 @@ void Provider_StatusUpdated(object sender, SubmoduleStatusEventArgs e)
             }
         }
 
-        public static void UpdateSubmoduleStatusAndWaitForResult(ISubmoduleStatusProvider provider, GitModule module, IReadOnlyList<GitItemStatus> gitStatus)
+        public static async void UpdateSubmoduleStatusAndWaitForResultAsync(ISubmoduleStatusProvider provider, GitModule module, IReadOnlyList<GitItemStatus> gitStatus)
```

Comment:
`async void` is evil.
```suggestion
        public static async Task UpdateSubmoduleStatusAndWaitForResultAsync(ISubmoduleStatusProvider provider, GitModule module, IReadOnlyList<GitItemStatus> gitStatus)
```

When you have some spare time, have a watch https://www.youtube.com/watch?v=yhmiWhB9Qns